dev/CommonStyles/AppBarSeparator.xaml

```diff
@@ -0,0 +1,63 @@
+<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
+    <ResourceDictionary.ThemeDictionaries>
+        <ResourceDictionary x:Key="Default">
+            <StaticResource x:Key="AppBarSeparatorForeground" ResourceKey="SystemControlForegroundBaseMediumLowBrush" />
+        </ResourceDictionary>
+        <ResourceDictionary x:Key="Light">
+            <StaticResource x:Key="AppBarSeparatorForeground" ResourceKey="SystemControlForegroundBaseMediumLowBrush" />
+        </ResourceDictionary>
+        <ResourceDictionary x:Key="HighContrast">
+            <StaticResource x:Key="AppBarSeparatorForeground" ResourceKey="SystemControlForegroundBaseMediumLowBrush" />
+        </ResourceDictionary>
+    </ResourceDictionary.ThemeDictionaries>
+
+    <Thickness x:Key="AppBarSeparatorMargin">16,12,15,12</Thickness>
+    <Thickness x:Key="AppBarOverflowSeparatorMargin">12,4,12,4</Thickness>
+    <x:Double x:Key="AppBarSeparatorWidth">1</x:Double>
+    <x:Double x:Key="AppBarOverflowSeparatorHeight">1</x:Double>
+    
+    <Style TargetType="AppBarSeparator" BasedOn="{StaticResource DefaultAppBarSeparatorStyle}"/>
+
+    <Style x:Key="DefaultAppBarSeparatorStyle" TargetType="AppBarSeparator">
+        <Setter Property="Foreground" Value="{ThemeResource AppBarSeparatorForeground}" />
+        <Setter Property="IsTabStop" Value="False" />
+        <Setter Property="Template">
+            <Setter.Value>
+                <ControlTemplate TargetType="AppBarSeparator">
+                    <Grid x:Name="RootGrid">
+                        <VisualStateManager.VisualStateGroups>
+                            <VisualStateGroup x:Name="ApplicationViewStates">
+                                <!-- FullSize is used when we are in landscape or filled mode -->
+                                <VisualState x:Name="FullSize"/>
+                                <!-- Compact is used when we are in portrait or snapped mode -->
+                                <VisualState x:Name="Compact">
+                                    <VisualState.Setters>
+                                        <Setter Target="RootGrid.Height" Value="{ThemeResource AppBarThemeCompactHeight}"/>
+                                        <Setter Target="RootGrid.VerticalAlignment" Value="Top"/>
+                                    </VisualState.Setters>
+                                </VisualState>
+                                <VisualState x:Name="Overflow">
+                                    <VisualState.Setters>
+                                        <Setter Target="SeparatorRectangle.Width" Value="NaN"/>
+                                        <Setter Target="SeparatorRectangle.HorizontalAlignment" Value="NaN"/>
+                                        <Setter Target="SeparatorRectangle.Height" Value="{StaticResource AppBarOverflowSeparatorHeight}"/>
```

Comment:
Responding to your PR description on height animations: have you tried using DoubleAnimationsUsingKeyFrames with something like `StoryBoard.TargetProperty="(UIElement.RenderTransform).(CompositeTransform.ScaleY)"`?

Reply:
How would we be able to use scaleY here?  We only know the height in compact mode, we don't really know the height when expanded nor the scale we need to "collapse" the separator.

Reply:
Also given that the labels for the appbar buttons "snap out" instead of fading away, having the separator also jump seems more consistent here. I think revisiting the whole animation experience of this control as @mdtauk suggested would be a better place to add the animation.

Reply:
Ah, I misinterpreted the implementation here. I think the separator jumping is quiet jarring, but I agree that the whole animation experience can be revisited. @YuliKl should this be added to the backlog?

Reply:
Labels fading as they slide up or down would be a nice touch.  The animation for the secondary menu items should probably feel familiar to the CommandBarFlyout control, as it is on the surface, the same control, but one is "docked" and the other is free-floating.

Reply:
I'll defer to @stmoy about animations.

Reply:
I think animations would be a worthwhile thing to address, but addressing animations is a separate issue from the one here. In the spirit of "separation of concerns", I think animations should be tackled separately. I'll spin up a thread with our motion designer.

Reply:
@chingucoding @mdtauk - I followed up with our motion designer, and he proposed a similar solution as @mdtauk where the labels fade in/out as they slide up or down. He suggested the animation to occur over 83ms. 

@chingucoding - do you think this sort of animation is possible? Should I file a new issue, or is this small enough to handle directly via PR?

Reply:
If it's not urgent, I can try to do this in this PR, however I am not sure how easy this will be.

Reply:
Having looked into it a bit now, I was unable to get this thing to animate at all using DoubleAnimations; the animation just jumps to the last value after the duration has finished. Could it be that AppBar... is switching states with `useTransitions` set to false? Given that CommandBar is in the OS as of right now, there is no way for me to check nor to change that so it might be best to create a tracking issue to add animations to this with WinUI 3 when we can change that (if my assumptions are correct).

Following XAML code reproduces this behavior:
![Image showing choppy animation of AppBarSeparator](https://user-images.githubusercontent.com/16122379/104054237-081ab880-51ed-11eb-9340-5917dd4ff1e8.gif)

```xaml
<VisualStateGroup.Transitions>
    <VisualTransition From="FullSize" To="Compact">
        <Storyboard>
            <DoubleAnimation Duration="0:0:0.500"
                To="{ThemeResource AppBarThemeCompactHeight}"
                Storyboard.TargetName="RootGrid"
                Storyboard.TargetProperty="Height"
                />
        </Storyboard>
    </VisualTransition>
    <VisualTransition From="Compact" To="FullSize">
        <Storyboard>
            <DoubleAnimation Duration="0:0:1.500"
                From="{ThemeResource AppBarThemeCompactHeight}"
                To="100"
                Storyboard.TargetName="RootGrid"
                Storyboard.TargetProperty="Height"/>
        </Storyboard>
    </VisualTransition>
</VisualStateGroup.Transitions>
```

Reply:
I think there are two separate animations to consider:
- The height of the separator animation
- The opacity/fade animation of the labels

My note above was about the second, not the first. I don't think the separator snapping in is as jarring as the labels being clipped.

That said, I'm pretty sure the issue with the markup shown above is that it's trying to animate **height** which is a no-no. Even outside CommandBar, I don't know if that markup would work. Another idea to try out would be to animate the scale of the separator along the Y-direction, which I think might be possible (but would also likely introduce a new RenderTransform if one doesn't already exist).

Reply:
Oh I thought we would want to do both. So just to clarify, we will only want to have a fadein/fadeout for the AppBarButton labels? 

Regarding the height animation: 
There are multiple problems here I think. The double animation should work in my opinion (why would it not, height is just a property after all). 
I also tried using Scale-Y but that also did not seem to work as that just crashes, below is the XAML I used:
```xaml
<VisualTransition From="Compact" To="FullSize">
    <Storyboard>
        <DoubleAnimation Duration="0:0:1.500"
            From="1"
            To="2"
            Storyboard.TargetName="RootGrid"
            Storyboard.TargetProperty="(UIElement.RenderTransform).(CompositeTransform.ScaleY)"/>
    </Storyboard>
</VisualTransition>
`` 
But even if that would work, the bigger issue is that we don't know what the start/end position would be since we don't know the actual end height in full size (this might be customized using the resources or labels might take a lot of space). That's why I tried to opt for height since we at least know a few things for that property.

But I'll try to get the fadein/fadeout animation working. Regarding the height animation of the separator, should we create a tracking issue then?

Reply:
I was able to add an animation to the AppBarButton though:

![Gif showing fade in/fade out animation on AppBarButton label](https://user-images.githubusercontent.com/16122379/104232173-41069780-5450-11eb-9471-4c4ab5db5ce1.gif)

The animation is very subtle and it doesn't look that different compared to the normal collapsing of the label. The left button has an opacity animation with duration 0:0:0.083, the right uses Visibility.Collapsed/Visible. 

@stmoy @mdtauk Do you think this is good to add to this PR? Or should we tinker with this a bit more?

Reply:
Maybe it's a timing thing, the opacity fade out could start just before the label slides down, and the fade in could end just after the label slides into position.

Could be done with easing curves maybe?

Reply:
I don't think the fade out timing can be changed without WinUI 3. It seems that the animation starts WAY to late (I think when it reaches the bottom) so we are out of luck with that here. We could add a delay for the fade in, but not sure what the appropriate delay would be, especially as it doesn't seem to bad to me right now. I'm concerned about the fade out here which looks almost identical to the label just being collapsed.

Reply:
Thanks for the investigation, @chingucoding - I agree that the animation in the GIF above doesn't seem to be timed correctly and even though I see the animation subtly on both the in and the out, I don't think it's prominent enough to justify making the change. I wonder if there is some coordination thing happening where the animation doesn't play until the transition is complete or something...

In any case, it seems like this wasn't as easy as I'd originally hoped, so I'm ok marking this as a "we'll deal with this later" issue.

Reply:
Sounds like a good plan, let's revisit this later. I think this PR is good go go now given that all feedback was addressed I think.